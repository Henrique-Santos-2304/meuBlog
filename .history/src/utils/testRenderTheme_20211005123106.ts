import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
