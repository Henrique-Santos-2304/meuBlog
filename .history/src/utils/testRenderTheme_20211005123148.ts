import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
