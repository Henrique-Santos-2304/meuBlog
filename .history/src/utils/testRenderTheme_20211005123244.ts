import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
