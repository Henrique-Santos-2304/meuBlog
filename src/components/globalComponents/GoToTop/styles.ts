import styled from "styled-components";
import { FiChevronUp } from "react-icons/fi";
export const Wrapper = styled.div`
  position: fixed;
  bottom: 2rem;
  right: 3rem;
  background: rgba(0, 0, 0, 0.65);
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  transition: all 0.5s linear;
  &:hover {
    border: 0.1rem solid ${({ theme }) => theme.colors.primary};
    filter: brightness(0.9);
  }
  @media (min-width: ${({ theme }) => theme.media.medium}) {
    bottom: 4rem;
    right: calc(50% - 30rem);
    transform: translateX(-50%);
  }
`;
export const Anchor = styled.a`
  display: flex;
  align-items: center;
  justify-content: center;
`;
export const Icon = styled(FiChevronUp)`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 3.2rem;
`;
