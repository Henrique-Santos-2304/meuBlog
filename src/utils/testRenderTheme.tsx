import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
