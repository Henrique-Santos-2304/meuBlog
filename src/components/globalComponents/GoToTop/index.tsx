import * as S from "./styles";

const GoToTop = () => (
  <S.Wrapper title="Voltar ao Inicio">
    <S.Anchor href="#">
      <S.Icon />
    </S.Anchor>
  </S.Wrapper>
);

export default GoToTop;
