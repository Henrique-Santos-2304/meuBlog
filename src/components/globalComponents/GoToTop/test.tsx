import GoToTop from ".";
import { renderTheme } from "utils/testRenderTheme";

describe("<GoToTop />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<GoToTop />);
    expect(container).toMatchSnapshot();
  });
});
