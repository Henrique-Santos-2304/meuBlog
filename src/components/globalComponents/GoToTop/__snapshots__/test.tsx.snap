// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoToTop /> should render  1`] = `
.c0 {
  position: fixed;
  bottom: 2rem;
  right: 3rem;
  background: rgba(0,0,0,0.65);
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  z-index: 100;
}

.c0:hover {
  border: 0.1rem solid #f8f8f8;
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c2 {
  color: #f8f8f8;
  font-size: 3.2rem;
}

@media (min-width:620px) {
  .c0 {
    bottom: 4rem;
    right: calc(50% - 30rem);
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
  }
}

<div>
  <div
    class="c0"
    title="Voltar ao Inicio"
  >
    <a
      class="c1"
      href="#"
    >
      <svg
        class="c2"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="18 15 12 9 6 15"
        />
      </svg>
    </a>
  </div>
</div>
`;
